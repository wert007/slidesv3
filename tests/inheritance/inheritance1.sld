abstract struct Parent {
    field1: int;

    func sayHello() {
        print('Hello ' + this.field1);
    }
}

struct Child: Parent {
    func $toString() -> string {
        base.sayHello();
        return 'child' + base.field1;
    }
}

// struct Child: Parent {
//     func $toString() -> string {
//         return 'child' + base;
//     }
// }

func main() {
    let p = new Child(42);
    print(p);
    p.field1 = 4;
    p.sayHello();
}