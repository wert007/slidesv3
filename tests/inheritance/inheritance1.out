Hello 42
child42
Hello 4
struct Cat {
    v = 4,
}
= 0
